datafusion/sql/src/unparser/expr.rs

```diff
@@ -1184,10 +1338,7 @@ impl Unparser<'_> {
                             fractional_seconds_precision: None,
                         };
                         Ok(ast::Expr::Interval(interval))
-                    } else if v.months == 0
-                        && v.days >= 0
-                        && v.nanoseconds % 1_000_000 == 0
-                    {
+                    } else if v.months == 0 && v.nanoseconds % 1_000_000 == 0 {
```

Comment:
This is intentional change (removed `v.days >= 0`) to fix interval generation for `IntervalStyle::SQLStandard` and negative days, the following  additional test was added for SQLStandard (didn't work previously)

```rust
            (
                interval_month_day_nano_lit("-1.5 DAY"),
                IntervalStyle::SQLStandard,
                "INTERVAL '-1 -12:0:0.000' DAY TO SECOND",
            ),
```